src/js/App/GlobalFilter/GlobalFilter.js

```diff
@@ -1,18 +1,131 @@
-import React, { useEffect, Fragment, useState, useCallback } from 'react';
+import React, { useEffect, Fragment, useState, useCallback, memo } from 'react';
+import PropTypes from 'prop-types';
 import { useSelector, useDispatch, batch, shallowEqual } from 'react-redux';
-import { GroupFilter } from '@redhat-cloud-services/frontend-components/components/cjs/ConditionalFilter';
-import { useTagsFilter } from '@redhat-cloud-services/frontend-components/components/cjs/FilterHooks';
-import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/components/cjs/Skeleton';
+import { GroupFilter } from '@redhat-cloud-services/frontend-components/ConditionalFilter';
+import { useTagsFilter } from '@redhat-cloud-services/frontend-components/FilterHooks';
+import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/Skeleton';
 import { fetchAllSIDs, fetchAllTags, fetchAllWorkloads, globalFilterChange } from '../../redux/actions';
-import { Split, SplitItem } from '@patternfly/react-core/dist/js/layouts/Split';
-import { Chip, ChipGroup } from '@patternfly/react-core/dist/js/components/ChipGroup';
-import { Button } from '@patternfly/react-core/dist/js/components/Button';
-import { Tooltip } from '@patternfly/react-core/dist/js/components/Tooltip';
+import { Button, Chip, ChipGroup, Split, SplitItem, Tooltip } from '@patternfly/react-core';
 import TagsModal from './TagsModal';
 import { workloads, updateSelected, storeFilter, generateFilter } from './constants';
 import debounce from 'lodash/debounce';
 import { useHistory } from 'react-router-dom';
 
+const GlobalFilterDropdown = ({
+  isAllowed,
+  isDisabled,
+  userLoaded,
+  filter,
+  chips,
+  setValue,
+  selectedTags,
+  isOpen,
+  filterTagsBy,
+  filterScope,
+  setIsOpen,
+}) => {
+  const dispatch = useDispatch();
+  const GroupFilterWrapper = !isAllowed() || isDisabled ? Tooltip : ({ children }) => <Fragment>{children}</Fragment>;
```

Comment:
Wouldn't it be better to use state or useCallback for this? The `disabled` state won't change that much and this is calculated on each render.

Reply:
Yup we need memo not callback here